cub/cub/device/dispatch/dispatch_transform.cuh

```diff
@@ -174,20 +174,17 @@ struct dispatch_t<StableAddress,
   KernelSource kernel_source             = {};
   KernelLauncherFactory launcher_factory = {};
 
-#ifdef _CUB_HAS_TRANSFORM_UBLKCP
-  // TODO(bgruber): I want to write tests for this but those are highly depending on the architecture we are running
-  // on?
-  template <typename ActivePolicy>
-  CUB_RUNTIME_FUNCTION _CCCL_VISIBILITY_HIDDEN _CCCL_FORCEINLINE auto configure_ublkcp_kernel() -> cuda_expected<
+  template <typename ActivePolicy, typename SMemFunc>
+  CUB_RUNTIME_FUNCTION _CCCL_VISIBILITY_HIDDEN _CCCL_FORCEINLINE auto
+  configure_async_kernel([[maybe_unused]] int alignment, SMemFunc smem_for_tile_size) -> cuda_expected<
     ::cuda::std::
       tuple<THRUST_NS_QUALIFIER::cuda_cub::detail::triple_chevron, decltype(kernel_source.TransformKernel()), int>>
   {
+    // Benchmarking shows that even for a few iteration, this loop takes around 4-7 us, so should not be a concern.
     using policy_t          = typename ActivePolicy::algo_policy;
     constexpr int block_dim = policy_t::block_threads;
-    _CCCL_ASSERT_HOST(block_dim % bulk_copy_align == 0,
-                      "block_threads needs to be a multiple of bulk_copy_alignment"); // then tile_size is a multiple of
-                                                                                      // it
-
+    _CCCL_ASSERT(block_dim % alignment == 0,
```

Comment:
Nitpick: I find it confusing that the variable is named `block_dim` when it is otherwise referred to as `block_threads`

Reply:
@bernhardmgruber `block_dim` on its own refers to a multi-dimensional thing. What grep picks up is likely from `block_dim_x` and only makes sense in the presence of at least `_y`. `block_threads` is inherently one-dimensional, so it's a bit easier to parse, since it means the total number of threads in a block and doesn't raise a question about the dimensionality of this variable.

Reply:
I replaced `block_dim` by `block_threads` in the transform dispatcher and kernels.